### DISCUSSION TITLE
Theming

### DISCUSSION THREAD

**Discussion by retailcoder** [rubberduck-vba/Rubberduck3#82] [Ideas]
How do we go about theming?

Right now there's `LightTheme.xaml`, `DarkTheme.xaml`, and `AbyssTheme.xaml`, each broken in varying degrees, but statically loaded into a merged dictionary in `DefaultStyle.xaml` (carried from RD2)... Which means the theme can't be configurable at run-time... which won't do.

Storing themes as xaml can't be it; we'll want to deserialize themes from settings, so we're looking at a JSON format. With how RD3 settings work, it means a record class (with a name and a description and a default value) for each configurable color.

We can get this far without thinking too hard about it, but there's the matter of _exactly what_ a theme should be defining, and how these colors are going to be named.

There are probably frameworks available, worth looking into not reinventing that wheel too much; Dragablz is known to be compatible with some, but if it doesn't work well with what we go with we can always tweak Dragablz styles and templates to meet our needs.

**Comment by Greedquest** [comment]
Themes should work like when you set the document theme (fonts and colour palette) in excel or word (ok probably not because no other code editors do that, but it would be funny. And intuitive)
